injected/src/features/message-bridge/create-page-world-bridge.js

```diff
@@ -0,0 +1,134 @@
+import {
+    InstallProxy,
+    ProxyNotification,
+    ProxyRequest,
+    ProxyResponse,
+    SubscriptionRequest,
+    SubscriptionResponse,
+    SubscriptionUnsubscribe,
+} from './schema.js';
+
+/**
+ * todo: use the already-captured ones
+ */
+const captured = {
+    dispatchEvent: globalThis.dispatchEvent.bind(globalThis),
+    addEventListener: globalThis.addEventListener.bind(globalThis),
+    removeEventListener: globalThis.removeEventListener.bind(globalThis),
+    CustomEvent: globalThis.CustomEvent,
+    Promise: globalThis.Promise,
+    String: globalThis.String,
+    mathRandom: globalThis.Math.random.bind(globalThis.Math),
+};
+
+/**
+ * This part can be called from any script.
+ * @param {string} featureName
+ * @param {string} token
+ * @return {Pick<import("@duckduckgo/messaging").Messaging, 'notify' | 'subscribe' | 'request'>}
+ */
+export function createPageWorldBridge(featureName, token) {
+    console.log('will createPageWorldBridge for', featureName, 'with token', token);
+
+    /**
+     * TODO: what is the value of this token? - since 'createPageWorldBridge' is available to all page scripts, probably none
+     * @param {string} eventName
+     * @return {`${string}-${string}`}
+     */
+    function appendToken(eventName) {
+        return `${eventName}-${token}`;
+    }
+
+    /**
+     * @param {{name: string} & Record<string, any>} incoming
+     */
+    function send(incoming) {
+        const event = new captured.CustomEvent(appendToken(incoming.name), { detail: incoming });
+        captured.dispatchEvent(event);
+    }
+
+    function random() {
+        return captured.String(captured.mathRandom());
+    }
+
+    // first, install the proxy
+    send(new InstallProxy({ featureName }));
+
+    /**
+     * return the `Messaging` interface for consumers
+     * Note:
+     */
+    return {
```

Comment:
Can this be a class definition to make the code more left bound?

Reply:
I don't care about style here, but having an explicit type for this (instead of `Pick<...>`) would be neat.